### ISSUE TITLE
in Mac  system, I can not  change the number of core in parallel setting

### ISSUE THREAD

**Issue by RezaTalebloo** [monty-se/PINstimation#5]
hi
thanks for very nice and useful package

I use this command in MacBookpro :

 options(pinstimation.parallel.cores = 6)
 getOption("pinstimation.parallel.cores")

but still computation is based on 2 cores!

(my laptop config:  2.3 GHz 8-Core Intel Core i9)

**Comment by monty-se** [comment]
Hi Reza,

Thanks for the nice words! There was indeed an issue with setting the number of cores in parallel computing. I have set an upper bound of 2 cores required for CRAN tests, and did not lift it. The commit f8e9832 has now solved the issue, so there should be no problem in setting the number of parallel cores higher than 2. In order to use this new feature, you need to download the development version of the package from GitHub. Follow these steps:
1. Remove the installed version of PINstimation
```R
remove.packages("PINstimation")
```` 
2- Install (and load) the package `devtools` if it is not already installed in your machine.
```R
install.packages("devtools")
library(devtools)
```` 
3- Install (and load) the development version of `PINstimation` from GitHub.
```R
devtools::install_github("monty-se/PINstimation", build_vignettes = TRUE)
library(PINstimation)
```` 

All should work fine now. Please let me know if any issues persist.

**Comment by RezaTalebloo** [comment]
thanks for reply,

now there is new problem in  usage.
 I remove and install the package and estimate the model but I receive this error:

|[=] Classification algorithm 	: EMO algorithm
  |[=] Number of trades in dataset	: 3 865 trades
  |[=] Time lag of lagged variables 	: 1000 milliseconds
  |[1] Computing lagged variables 	: using parallel processing
  |                                     |   0% of variables computedError in eval(first, envir = parent.frame(), enclos = baseenv()) : 
  object 'multisession' not found
Called from: eval(first, envir = parent.frame(), enclos = baseenv())

Thank you for helping me solve this problem.

**Comment by monty-se** [comment]
Reza, is it a related problem or a new one? does the problem occur when the number of cores is set to 2 - as it used to be?

**Comment by RezaTalebloo** [comment]
you are right.
This problem is different from the issue of using the number of cores, but it actually appeared after the reinstallation, because for the past few days I have easily used the software package and had no problems with it, and this problem just appeared after new change . In any case, if you think it is advisable to close this issue on this page and open this problem in a new issue?

**Comment by RezaTalebloo** [comment]
<img width="875" alt="Screen Shot 2023-08-04 at 5 31 14 PM" src="https://github.com/monty-se/PINstimation/assets/46419013/37093ece-b860-4639-a7aa-2029db84f899">

**Comment by monty-se** [comment]
I have installed the development version, experimented with the number of cores, the different algorithms and timelags but do not seem to have any issues on my computer. Can you make the regular checks? - restart the R session - Reinstall the CRAN version and try the same instructions - Delete the CRAN version and then install the development version and then try again.

**Comment by RezaTalebloo** [comment]
Thank you very much for your time. The problem was not with the "PINstimation" and it was solved. Thank you very much for your help.
